core/camera/src/main/java/com/google/jetpackcamera/core/camera/CameraXCameraUseCase.kt

```diff
@@ -621,6 +627,44 @@ constructor(
         } ?: this
     }
 
+    private fun CameraAppSettings.tryApplyFrameRateConstraints(): CameraAppSettings {
+        return systemConstraints.perLensConstraints[cameraLensFacing]?.let { constraints ->
+            with(constraints.supportedFixedFrameRates) {
+                val newTargetFrameRate = if (contains(targetFrameRate)) {
+                    targetFrameRate
+                } else {
+                    TARGET_FPS_AUTO
+                }
+
+                this@tryApplyFrameRateConstraints.copy(
+                    targetFrameRate = newTargetFrameRate
+                )
+            }
+        } ?: this
+    }
+
+    private fun CameraAppSettings.tryApplyStabilizationConstraints(): CameraAppSettings {
```

Comment:
It looks like there are some constraints that need to be handled here that have to do with the fps. They're currently encoded in `CameraXCameraUseCase.shouldVideoBeStabilized()` and `CameraXCameraUseCase.shouldPreviewBeStabilized()`.